src/mppt_algorithms/mppt.py

```diff
@@ -87,7 +87,7 @@ def iterate(self, v_in, i_in, t_in, cycle):
         """
         return self.v_ref
 
-    def calc_perturb_amt(self, v_ref, v_in, i_in, t_in):
+    def calc_perturb_amt(self, v_ref, v_in, i_in, t_in, p_old, v_old):
```

Comment:
Note that we include p_old and v_old on line 28 of this file as a class instance property; we can calculate power with v_in * i_in and store it in p_old and v_old at the end of the function.